### PR TITLE
fix: fetch image to show

### PR DESCRIPTION
**Pull request by lelenei**
Closes mf-799

### CODE REVIEW

File: packages/plugins/RSS3/src/SNSAdaptor/components/ImageHolder.tsx [hunk 1]
```diff
@@ -12,9 +15,19 @@ export const ImageHolder = ({ url, title, isFullRound, size }: ImageProps) => {
...
+
     return (
         <div className={`flex justify-around relative ${roundedClass}`} style={containerStyles}>
-            <img className={`object-cover w-full h-full ${roundedClass}`} src={url} alt={title} />
+            {loading ? (
+                <CircularProgress size="small" />
+            ) : (
+                <img className={`object-cover w-full h-full ${roundedClass}`} src={image} alt={title} />
+            )}
         </div>
```
Comment (guanbinrui) [inline packages/plugins/RSS3/src/SNSAdaptor/components/ImageHolder.tsx]: Replace it with `packages/shared/src/UI/components/Image/index.tsx`.

File: packages/shared/src/UI/components/Image/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { useAsync } from 'react-use'
+
+export function Image(props: ImgHTMLAttributes<HTMLImageElement>) {
+    const { loading, value } = useAsync(async () => {
+        if (!props.src) return
+        const data = await globalThis.r2d2Fetch(props.src)
+        return URL.createObjectURL(await data.blob())
+    }, [props.src])
+
+    return <>{loading ? <CircularProgress size="small" /> : <img {...props} src={value ?? props.src} />}</>
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/Image/index.tsx]: It's better if we do some error handling here.

File: packages/shared/src/UI/components/Image/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { useAsync } from 'react-use'
+
+const useStyles = makeStyles()(() => ({}))
+
+interface ImageProps extends ImgHTMLAttributes<HTMLImageElement>, withClasses<'loadingFailImage'> {}
+
+const assetPlayerFallbackImageDark = new URL('./nft_token_fallback_dark.png', import.meta.url)
+const assetPlayerFallbackImageLight = new URL('./nft_token_fallback.png', import.meta.url)
+
+export function Image(props: ImageProps) {
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/Image/index.tsx:12]:
```suggestion
export function Image(props: ImageProps & { fallbackSrc: string | URL }) {
```

**Review by guanbinrui — APPROVED** [review]

File: packages/shared/src/UI/components/Image/index.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import type { ImgHTMLAttributes } from 'react'
+import { useAsync } from 'react-use'
+
+const useStyles = makeStyles()(() => ({}))
+
+interface ImageProps extends ImgHTMLAttributes<HTMLImageElement>, withClasses<'loadingFailImage'> {
+    fallbackImage?: URL
+}
+
+export function Image(props: ImageProps) {
```
Comment (Jack-Works) [inline packages/shared/src/UI/components/Image/index.tsx:12]: @lelenei duplicate with packages/mask/src/components/shared/Image.tsx

**Pull request merged**